src/runtime/recipe/recipe.ts

```diff
@@ -33,7 +33,7 @@ export type CloneMap = Map<RecipeComponent, RecipeComponent>;
 export type VariableMap = Map<TypeVariableInfo|Schema, TypeVariableInfo|Schema>;
 
 export type IsResolvedOptions = {showUnresolved?: boolean, details?: string[]}; // TODO(lindner): standardize details
-export type IsValidOptions = {errors?: Map<Recipe | RecipeComponent, string>};
+export type IsValidOptions = {errors?: Map<Recipe | RecipeComponent, string>, typeErrors?: string[]}; // TODO(cypher1): Fixup
```

Comment:
Fix up how?

Reply:
Thanks, I'd thought I could remove the `typeErrors` field (putting the errors back into the error map when they 'bubbled' up to a place where we knew what RecipeComponent they were associated with, unfortunately the typechecker doesn't always know where in the recipe it is operating so we can't.

I'll remove the comment, I don't think it's achievable.